applications/RomApplication/python_scripts/structural_mechanics_analysis_rom.py

```diff
@@ -32,8 +32,8 @@ def ModifyInitialGeometry(self):
             data=json.load(f)
             counter = 0
             rom_dofs=self.project_parameters["solver_settings"]["rom_settings"]["number_of_rom_dofs"].GetInt()
-            Dimensions = len(data['1.0'])  #Checking for the number of DOFs (a vector or a matrix is imported to each node)
             for node in computing_model_part.Nodes:
+                Dimensions = len(data[str(float(node.Id))])
```

Comment:
the input file should now contain the list of variables contained in the input and their ordering.

one should get info about the lenght by reading that block once and then using it here. (note that doing it this way will also work as a "sanity check"

Reply:
I am also in favor of this idea and below too.

Reply:
done. Also the numbering is performed with integers rather than doubles